### ISSUE TITLE
Multimedia keys on Linux

### ISSUE THREAD

**Issue by boppreh** [boppreh/keyboard#20]
Linux doesn't seem to emit any events for multimedia keys. In Windows they were already weird because of their null scan code, but in Linux literally nothing happens.

This should be fixed somehow.

**Comment by ahogen** [comment]
> Linux doesn't seem to emit any events for multimedia keys.

Does this mean you can't capture a multimedia button press, or you can't send a key press?

Looking at an application where I want to send a multimedia key (not capture) so would this issue affect my use? Thanks!

**Comment by boppreh** [comment]
It's been a while, let me check...

**Comment by boppreh** [comment]
TL;DR: Yes, it works, as long as you know the scan code.

I just tried with a Fedora 25 live CD.

Pressing `FN+F5` (brightness down in my laptop keyboard) results in Fedora muting the microphone. Not our fault here, but just to let you know.

`keyboard` correctly captured that key press with scan code of 190, even if the name was "unknown". You can check this by running `python -m keyboard`, which should print each key event as a small JSON object. And even better, the scan code was sufficient to reproduce the event.

This allowed me to record and replay the key presses, with the microphone correctly muting and unmuting as with a normal key press:

```
# python -m keyboard > events.txt
[...recording...]

# python -m keyboard < events.txt
[...replaying...]
```

If you are trying to emulate media keys that you don't have, I don't know what would happen. If you do try, please report back here and I'll note down for future users.

**Comment by ahogen** [comment]
Okay great. I'll go ahead and mess around with it then. Thanks for doing that little bit of research! I appreciate it!

I was looking at some of the Windows keyboard [codes here](https://msdn.microsoft.com/en-us/library/dd375731), with the volume up/down/mute buttons being of particular interest. 

| Constant/value | Description
| ---------------- | ------------- |
| VK_VOLUME_MUTE<br>0xAD | Volume Mute key |
| VK_VOLUME_DONE<br>0xAE | Volume Down key |
| VK_VOLUME_UP<br>0xAF | Volume Up key |
| VK_MEDIA_NEXT_TRACK<br>0x80 | Next track key |
| VK_MEDIA_PREV_TRACK<br>0x81 | Previous Track key |
| | |
| VK_MEDIA_PLAY_PAUSE<br>0x83 | Play/Pause Media key |

And then there are the keyboard/keypad codes defined by the USB Implementers Forum [here (pg. 56)](http://www.usb.org/developers/hidpage/Hut1_12v2.pdf).

| Usage ID<br>(Dec) | Usage ID<br>(Hex) | Usage Name |
| --------------------- | ---------------------- | -------------- |
| 127                          | 0x7F                          | Mute |
| 128                          | 0x80                         | Volume Up |
| 129                          | 0x81                         | Volume Down |

Maybe between the two of those I get get something working.
